_inc/lib/class.jetpack-search-performance-logger.php

```diff
@@ -59,20 +59,21 @@ private function should_log_query( $query ) {
 	}
 
 	private function record_query_time( $duration, $was_jetpack_search ) {
-		$this->stats[] = array( $was_jetpack_search, $duration * 1000 );
+		$this->stats[] = array( $was_jetpack_search, intval( $duration * 1000 ) );
 	}
 
 	public function print_stats() {
 		$beacons = array();
 		if ( ! empty( $this->stats ) ) {
 			foreach( $this->stats as $stat ) {
 				$search_type = $stat[0] ? 'es' : 'mysql';
-				$beacons[] = "jetpack.search.{$search_type}.duration:{$stat[1]}|ms";
+				$beacons[] = "%22jetpack.search.{$search_type}.duration:{$stat[1]}|ms%22";
 			}
+
+			$encoded_json = '{%22beacons%22:[' . implode(',', $beacons ) . ']}';
+			$encoded_site_url = urlencode( site_url() );
+			$url = "https://pixel.wp.com/boom.gif?v=0.9&u={$encoded_site_url}&json={$encoded_json}";
+			echo '<img src="' . $url . '" width="1" height="1" style="display:none;" />';
```

Comment:
`$url` should be run through `esc_url()` for safety.

Reply:
The problem with this - and don't laugh - is that esc_url irretrievably breaks the URL and prevents the stats from showing up. This is because it discards the `{` (which, when urlencoded, can't be properly parsed by the boom.gif ingestor).

Reply:
Lame!